src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -183,6 +183,16 @@ private void OnProjectsChanged(ConnectionInformation connection, IEnumerable<Son
                 var projects = await this.host.SonarQubeService.GetAllProjectsAsync(connection.Organization?.Key,
                     cancellationToken);
 
+                if (string.IsNullOrEmpty(connection.Password.ToUnsecureString()) &&
```

Comment:
I'll add a comment when I am sure this works but the idea is to say we want to have the special code only when we hit the 10K issue, the project we want to bind to is not in the 10K projects we have fetched AND this happens only when opening an already bound solution.